libsolidity/formal/SMTChecker.cpp

```diff
@@ -1172,7 +1196,8 @@ void SMTChecker::createExpr(Expression const& _e)
 void SMTChecker::defineExpr(Expression const& _e, smt::Expression _value)
 {
 	createExpr(_e);
-	m_interface->addAssertion(expr(_e) == _value);
+	if (isSupportedType(*_e.annotation().type))
+		m_interface->addAssertion(expr(_e) == _value);
```

Comment:
I'm not sure there would be a warning here. Maybe we should add a warning right here. If the type is not supported at all (declaration and operations), there would be a warning when we create the expression. But the `FunctionType` case wouldn't issue a warning since its declaration is supported, but not the equality operator (that's why the if here).

Reply:
In the assignment case it would have given a warning and wouldn't reach this code. This happens for example when calling a function `h` with `(h)(2)`, where the expression `(h)` is assigned whatever `h` is, but since it's a function we can't use equality.

Reply:
Ok, current idea is to make sure the warning is given before calling `defineExpr`, such that anything that reaches `defineExpr` needs to be supported.

Reply:
So can you add a warning, please?